dbcon/mysql/ha_mcs_pushdown.cpp

```diff
@@ -826,7 +831,10 @@ create_columnstore_select_handler(THD* thd, SELECT_LEX* select_lex)
                 sel->where= conds;
 
                 if (isPS)
-                    sel->prep_where= conds;
+                {
+                    sel->prep_where= conds ? conds->copy_andor_structure(thd) : 0;
+                    in_subselect_rewrite(sel);
```

Comment:
We should check the return value of this in_subselect_rewrite() call and set unsupported_feature. See the block at line 859.

Reply:
Good catch. This is now fixed.